Lib/compileall.py

```diff
@@ -152,8 +152,14 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
                           "in combination with stripdir or prependdir"))
 
     success = True
-    if quiet < 2 and isinstance(fullname, os.PathLike):
+    if isinstance(fullname, os.PathLike):
```

Comment:
nit: os.fspath is happy to take `str` and `bytes`, so you don't need this check for fullname (for the other three, you need something to guard against None)

Reply:
You're right! The last three`os.fspath` calls can simply be moved below into each of the blocks handling the three directory arguments, and the first one shouldn't need the `isinstance` guard (letting `os.fspath` error on invalid types seems like a good thing here!)

Reply:
Sorry for the delay! This fell off my working pool. Made the changes :blush: